### PR TITLE
Added support for extra long Breadcrumb chains

### PR DESCRIPTION
**Pull request by sarthak-marwaha**
This shouldn't really happen that often but it might happen on narrow viewports
image

### CODE REVIEW

**Review by jlhughes — COMMENTED** [review]
Nice.

File: src/foam/u2/stack/BreadcrumbView.js [hunk 1]
```diff
@@ -4,41 +4,69 @@
...
+      name: 'collapseBreakpoint',
+      documentation: `If the number of breadcrumbs goes over this value they will start collapsing into a dropdown,
+      default value is sum of maxHead and maxTail + 1 to avoid a single breadcrumb collapsing into a dropdown `,
+      factory: function() { return this.maxHead + this.maxTail + 1; }
+    },
+    {
+      class: 'FObjectArray',
+      of: 'foam.core.Action',
+      name: 'actionArray',
+      value: []
```
Comment (kgrgreer) [inline src/foam/u2/stack/BreadcrumbView.js]: value:[] will cause different instances of this class to all share the same array. You need to use a factory: instead, but FObjectArray already does that, so you should just remove this line.

File: src/foam/u2/stack/BreadcrumbView.js [hunk 2]
```diff
@@ -4,41 +4,69 @@
...
     function render() {
       this.SUPER();
       var self = this;
       this.addClass(this.myClass('display'));
-      if ( this.stack && this.stack.stack_ ) { 
+      if ( this.stack && this.stack.stack_ ) {
         var navStack = this.stack.stack_.slice(this.stack.navStackBottom, this.stack.pos);
         var themeIcon = navStack.length == 1 ? 'back' : '';
-        navStack.map((v, i, _) =>{
+        navStack.map((v, i, _) => {
```
Comment (kgrgreer) [inline src/foam/u2/stack/BreadcrumbView.js]: Do you need to include the _ if you aren't going to tuse it?

File: src/foam/u2/stack/BreadcrumbView.js [hunk 3]
```diff
@@ -4,41 +4,68 @@
...
-  requires: ['foam.core.Action'],
+  requires: [
+    'foam.core.Action',
+    'foam.u2.view.OverlayActionListView',
+  ],
 
   css: `
   ^display {
-    display: inline-flex;
+    display: flex;
```
Comment (kgrgreer) [inline src/foam/u2/stack/BreadcrumbView.js]: sort

File: src/foam/u2/stack/BreadcrumbView.js [hunk 4]
```diff
@@ -4,41 +4,68 @@
...
     align-items: center;
+    flex-wrap: wrap;
   }
   ^slash{
     padding: 8px;
     vertical-align: middle;
   }
+  ^breadCrumb > * {
+    overflow: hidden;
```
Comment (kgrgreer) [inline src/foam/u2/stack/BreadcrumbView.js:30]: sort

**Pull request merged**